src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/src/components/advanced/AddressComponent.tsx

```diff
@@ -203,134 +244,170 @@ export function AddressComponent(props: IAddressComponentProps) {
+          <AddressLabel
+            labelKey={'address_component.care_of'}
+            id={`address_care_of_${id}`}
+            language={language}
+            required={required}
+            readOnly={readOnly}
+            labelSettings={labelSettings}
+          />
           <input
-            id={`address_care_of_${props.id}`}
-            className={classNames('form-control',
-              {
-                'validation-error': (allValidations.careOf.errors.length),
-                disabled: props.readOnly,
-              })}
+            id={`address_care_of_${id}`}
+            className={cn('form-control', {
+              'validation-error': allValidations.careOf.errors.length,
+              disabled: readOnly,
+            })}
             value={careOf}
             onChange={updateField.bind(null, AddressKeys.careOf)}
             onBlur={onBlurField.bind(null, AddressKeys.careOf, careOf)}
-            readOnly={props.readOnly}
+            readOnly={readOnly}
           />
-          {allValidations?.[AddressKeys.careOf] ?
-            renderValidationMessagesForComponent(allValidations[AddressKeys.careOf],
-              `${props.id}_${AddressKeys.careOf}`)
+          {allValidations?.[AddressKeys.careOf]
+            ? renderValidationMessagesForComponent(
+                allValidations[AddressKeys.careOf],
+                `${id}_${AddressKeys.careOf}`,
+              )
             : null}
         </>
-      }
+      )}
 
       <div className='address-component-postplace-zipCode'>
         <div className='address-component-zipCode'>
-          {renderLabel('address_component.zip_code', `address_zip_code_${props.id}`)}
+          <AddressLabel
+            labelKey='address_component.zip_code'
+            id={`address_zip_code_${id}`}
+            language={language}
+            required={required}
+            readOnly={readOnly}
+            labelSettings={labelSettings}
+          />
           <input
-            id={`address_zip_code_${props.id}`}
-            className={classNames('address-component-small-inputs', 'form-control',
+            id={`address_zip_code_${id}`}
+            className={cn(
+              'address-component-small-inputs',
+              'form-control',
               {
-                'validation-error': (allValidations.zipCode.errors.length),
-                disabled: props.readOnly,
-              })}
+                'validation-error': allValidations.zipCode.errors.length,
+                disabled: readOnly,
+              },
+            )}
             value={zipCode}
             onChange={updateField.bind(null, AddressKeys.zipCode)}
             onBlur={onBlurField.bind(null, AddressKeys.zipCode, zipCode)}
-            readOnly={props.readOnly}
-            required={props.required}
-            type='number'
+            readOnly={readOnly}
+            required={required}
+            inputMode='numeric'
```

Comment:
This does seem to some browsers that does not support inputMode (https://caniuse.com/input-inputmode), for instance only supported by safari on iphone. Is this something we are aware of?

Reply:
Hm, where can you see its only supported on iOS? It looks to be supported on all major mobile browsers on the chart 
![image](https://user-images.githubusercontent.com/2082481/151939801-d06a7718-362e-42f9-b1ab-11c1327fe241.png)
. Theres also a great demo site here https://inputmodes.com/where you can test the different ones. I've manually tested on my Android, and the numeric keyboard does appear.

Reply:
![image](https://user-images.githubusercontent.com/10416981/151940343-acca7927-471c-48e0-9a8b-2a0d1c44cbbb.png)
Yeah, that's what I'm trying to say, **only** supported on safari on iOS, seems like this is not supported in desktop safari.

Reply:
Which then opens up for filling out non numeric values I guess? Have not tested this

Reply:
Not sure if this is really a problem, and the numeric keyboard is a great improvement for mobile users. Just a heads up.

Reply:
Oh, right, we also validate the zip code
```
zipCode && !zipCode.match(/^\d{4}$/)
```

Well, then this is a non issue I guess.

Reply:
Ah, I see what you mean now, sorry :) This is mainly to give a better experience on mobile devices, so the numeric keyboard appears by default. For browsers that doesnt support this, it will be ignored, and just behave as a regular `input=text` input.

I changed it from `input=number` because that is only really meant for inputting numbers, and zipcodes are not really numbers since they can start with 0. Additionally you get up/down arrows to increment/decrement the number and if you were to click these buttons and the zipcode starts with a 0, it will be stripped.

Relying on the input type to restrict to only numbers is not something that we should rely on I think. We have additional checks on blur that ensures only numbers are valid input.

Reply:
Didnt see all your other replies until now 😅 But looks like we agree then :)